File: DAY-3/CountNotificationCount.java
import java.util.*;
public class CountNotificationCount {
    public static List<Map<String, Object>> countNotifications(List<String> notifications) {
        Map<String, Integer> map = new HashMap<>();

       // Count occurrences of each notification category
       for (String val : notifications) {
           map.put(val, map.getOrDefault(val, 0) + 1);
       }

       // Convert the map to a list of objects
       List<Map<String, Object>> result = new ArrayList<>();
       for (Map.Entry<String, Integer> entry : map.entrySet()) {
           Map<String, Object> item = new HashMap<>();
           item.put("category", entry.getKey());
           item.put("count", entry.getValue());
           result.add(item);
       }

       // Sort alphabetically by category
       result.sort(Comparator.comparing(o -> (String) o.get("category")));
       return result;
   }
}

File: DAY-3/CountNotificationCount.js
function countNotifications(notifications) {
    let map = new Map();
    for(val of notifications) {
        if(map.has(val)) {
            map.set(val, map.get(val) + 1);
        }   else {
            map.set(val, 1);
        }
    }
    let array = [];
    map.forEach((val, key) => {
        array.push({
            "category": key,
            "count": val
        })
    })
    array.sort((a, b) => {
        return a.category.localeCompare(b.category);
    })
    return array;
  }